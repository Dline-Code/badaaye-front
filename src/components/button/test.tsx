/**
 * @jest-environment jsdom
 */
import React from 'react'
import { fireEvent, render } from '../../utils/tests/wrapper'
import Button from '.'

describe('ButtonComponent', () => {
  it('should render component correctly', () => {
    const { container, getByText } = render(
      <Button type="submit" children="Fazer Parte" />
    )

    expect(container.firstChild).toMatchSnapshot()
    expect(getByText('Fazer Parte')).toBeInTheDocument()
  })

  it('should render and click button', () => {
    const mockFunction = jest.fn()
    const { getByText } = render(
      <Button
        type="submit"
        children="Fazer Parte"
        onClick={() => mockFunction()}
      />
    )
    fireEvent.click(getByText('Fazer Parte'))
    expect(mockFunction).toBeCalled()
  })
})
