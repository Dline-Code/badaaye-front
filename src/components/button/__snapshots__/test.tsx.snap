// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent should render component correctly 1`] = `
<button
  class="styles__Button-sc-1x25beb-0 jeKdXn"
  type="submit"
>
  Fazer Parte
</button>
`;
